lib/memjs/memjs.d.ts

```diff
@@ -0,0 +1,531 @@
+   *
+   * If the argument is not given, fallback on the `MEMCACHIER_SERVERS` environment
+   * variable, `MEMCACHE_SERVERS` environment variable or `"localhost:11211"`.
+   *
+   * The options hash may contain the options:
+   *
+   * * `retries` - the number of times to retry an operation in lieu of failures
+   * (default 2)
+   * * `expires` - the default expiration in seconds to use (default 0 - never
+   * expire). If `expires` is greater than 30 days (60 x 60 x 24 x 30), it is
+   * treated as a UNIX time (number of seconds since January 1, 1970).
+   * * `logger` - a logger object that responds to `log(string)` method calls.
+   * * `failover` - whether to failover to next server. Defaults to false.
+   * * `failoverTime` - how much to wait until retring a failed server. Default
+   *                    is 60 seconds.
+   *
+   *   ~~~~
+   *     log(msg1[, msg2[, msg3[...]]])
+   *   ~~~~
+   *
+   *   Defaults to `console`.
+   *
+   * Or options for the servers including:
+   * * `username` and `password` for fallback SASL authentication credentials.
+   * * `timeout` in seconds to determine failure for operations. Default is 0.5
+   *             seconds.
+   * * 'conntimeout' in seconds to connection failure. Default is twice the value
+   *                 of `timeout`.
+   * * `keepAlive` whether to enable keep-alive functionality. Defaults to false.
+   * * `keepAliveDelay` in seconds to the initial delay before the first keepalive
+   *                    probe is sent on an idle socket. Defaults is 30 seconds.
+   */
+  static create(
+    serversStr?: string,
+    options?: ClientOptions | ServerOptions
+  ): Client;
+
+  servers: string[];
+
+  /**
+   * Client initializer takes a list of Servers and an options dictionary. See Client.create for details.
+   * @param servers
+   * @param options
+   */
+  constructor(servers: string, options?: ClientOptions);
+
+  /**
+   * Chooses the server to talk to by hashing the given key.
+   * @param key
+   */
+  server(key: string): string;
+
+  /**
+   * GET
+   *
+   * Retrieves the value at the given key in memcache.
+   *
+   * The callback signature is:
+   *
+   *     callback(err, value, flags)
+   *
+   * _value_ and _flags_ are both `Buffer`s. If the key is not found, the
+   * callback is invoked with null for both arguments and no error
+   * @param key
+   * @param callback
+   */
+  get(key: string): Promise<{ value: Buffer; flags: Buffer }>;
+  get(
+    key: string,
+    callback: (
+      err: Error | null,
+      value: Buffer | null,
+      flags: Buffer | null
+    ) => void
+  ): void;
+
+  /**
+   * MULTI-GET / GET-MULTI
+   *
+   * Retrieves the value at the given keys in memcache.
+   *
+   * The callback signature is:
+   *
+   *     callback(err, values, flags)
+   *
+   * @param keys
+   * @param callback
+   */
+  getMulti<Keys extends string>(
+    keys: Keys[]
+  ): Promise<{ values: { [K in Keys]?: Buffer | null }; flags: Buffer | null }>;
```

Comment:
you're sure? these really seem like strings to me based on the unittests, I think the docs in memjs had been wrong?

Reply:
```
  var assertor = function(err, val, flags) {
    console.log('types')
    console.error(val['hello1'])
    console.error(typeof val['hello1'])
    t.deepEqual({
      hello1: 'world1',
      hello3: 'world3',
      hello4: 'world4',
    }, val);
    t.false(flags);
    t.equal(null, err);
  };
  client.getMulti(['hello1', 'hello2', 'hello3', 'hello4'], assertor);
  testAllCallbacksEmpty(t, dummyServer1);
  testAllCallbacksEmpty(t, dummyServer2);

  return client.getMulti(['hello1', 'hello2', 'hello3', 'hello4']).then(function(res) {
    assertor(null, res.values, res.flags);
```

```
world1
string
world1
string
```

Reply:
Very weird. Here's me, in a nodejs REPL:
![image](https://user-images.githubusercontent.com/296279/112872804-a6dce580-9075-11eb-93f9-c49f6ebbac91.png)

Test script using promise:

![image](https://user-images.githubusercontent.com/296279/112873213-32ef0d00-9076-11eb-81e2-8071d0b14977.png)